include/effectmanager.h

```diff
@@ -482,6 +482,31 @@ class EffectManager : public IJSONSerializable
         return _vEffects[i]->IsEnabled();
     }
 
+    void MoveEffect(size_t from, size_t to)
+    {
+        if (from >= _vEffects.size() || to >= _vEffects.size())
+        {
+            debugW("Invalid index for MoveEffect");
+            return;
+        }
+
+        if (from == to)

```

Comment:
What about vector::move?  I've never used it but it feels like it could do this without rotating the others around?

Reply:
As I understand it, vector::move is used to move the items from one vector to another (that is, it prevents copying when migrating them).
vector::rotate is intended for the purpose we need here. And used as it is in this code it moves (pardon, rotates) exactly one effect, leaving all others alone.